File: frontend-react/src/services/organizationService.js
import api from './api';

const create = (data) => api.post('/organizations', data);
const getById = (id) => api.get(`/organizations/${id}`);
const update = (id, data) => api.put(`/organizations/${id}`, data);
const getStaff = (id) => api.get(`/organizations/${id}/staff`);
const addStaff = (id, data) => api.post(`/organizations/${id}/staff`, data);
const removeStaff = (id, uid) => api.delete(`/organizations/${id}/staff/${uid}`);

const organizationService = { create, getById, update, getStaff, addStaff, removeStaff };
export default organizationService;

File: src/main/java/com/example/kyrgyzstancentralmedicalcard/services/OrganizationService.java
package com.example.kyrgyzstancentralmedicalcard.services;

import com.example.kyrgyzstancentralmedicalcard.entity.Organization;
import com.example.kyrgyzstancentralmedicalcard.entity.OrganizationUser;

import java.util.List;

public interface OrganizationService {
    Organization create(Organization organization);
    Organization getById(Long id);
    Organization update(Long id, Organization organization);
    List<OrganizationUser> getStaff(Long organizationId);
    OrganizationUser addStaff(Long organizationId, OrganizationUser organizationUser);
    void removeStaff(Long organizationId, Long userId);
}
